### PR TITLE
Allow selection of gps information to show in toolbar

### PR DESCRIPTION
**Pull request by nyalldawson**
Allows selection of which GPS information should be shown in the floating GPS toolbar:

https://user-images.githubusercontent.com/1829991/200457732-693c280a-3bd0-45e0-8dcb-b5080d2c7348.mp4

Sponsored by NIWA

### CODE REVIEW

File: src/app/gps/qgsgpstoolbar.cpp [hunk 1]
```diff
@@ -322,3 +373,49 @@ void QgsGpsToolBar::destinationMenuAboutToShow()
   }
 }
 
+void QgsGpsToolBar::createLocationWidget()
+{
+  mLocationButton = new QToolButton();
+  mLocationButton->setToolTip( tr( "Current GPS Location" ) );
```
Comment (nirvn) [inline src/app/gps/qgsgpstoolbar.cpp:379]: You'll hate me  for suggesting this but shouldn't the button be named mInformationButton, and its tooltip be "Current GPS Location Information" ? That'd better reflect what this is about, and better match the appropriately named Qgis::GpsInformationComponent class name.

File: src/app/gps/qgsgpstoolbar.cpp [hunk 2]
```diff
@@ -322,3 +373,49 @@ void QgsGpsToolBar::destinationMenuAboutToShow()
...
+
+  QMenu *locationMenu = new QMenu( mLocationButton );
+
+  const Qgis::GpsInformationComponents visibleComponents = settingShowInToolbar.value();
+
+  for ( const auto &it : std::vector< std::pair< Qgis::GpsInformationComponent, QString> >
+{
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/app/gps/qgsgpstoolbar.cpp:386]: You're correct -- it's gross. But I can't get astyle to make it nicer :(

[the remark this answers is not in the document]
Comment (nirvn) [reply src/app/gps/qgsgpstoolbar.cpp:386]: @nyalldawson , sometimes astyle feels like

giphy

Alright, I suspected as much :)

File: src/app/gps/qgsgpstoolbar.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/gps/qgsgpstoolbar.cpp:379]: Done!

**Review by nirvn — APPROVED** [review]
@nyalldawson , thanks.

**Pull request merged**